lib/common/monitoring/alarms/DatabaseAlarmFactory.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ComparisonOperator,
+  TreatMissingData,
+} from "aws-cdk-lib/aws-cloudwatch";
+
+import { AlarmFactory, CustomAlarmThreshold } from "../../alarm";
+import { MetricWithAlarmSupport } from "../../metric";
+
+export interface LowConnectionCountThreshold extends CustomAlarmThreshold {
+  readonly minConnectionCount: number;
+}
+
+export interface HighConnectionCountThreshold extends CustomAlarmThreshold {
+  readonly maxConnectionCount: number;
+}
+
+export class DatabaseAlarmFactory {
```

Comment:
May be worth just making this generic towards the concept of connections. Not all databases have connection metrics (e.g. DynamoDB), and things with connections might not be databases (e.g. EC2 instances).

Reply:
Good callout, we could name it ConnectionAlarmFactory for now.